### ISSUE TITLE
USB modes (MTP,Network) don't work

### ISSUE THREAD

**Issue by rinigus** [sailfishos-sony-tama/main#91]
From https://talk.maemo.org/showpost.php?p=1570415&postcount=187

```
Dez 05 19:17:24 XperiaXZ3 mce[3140]: modules/battery-udev.c: mcebat_update(): charger_type: none -> usb
Dez 05 19:17:24 XperiaXZ3 usb_moded[3139]: /config/usb_gadget/g1/UDC: write failure: Value too large for defined data type
Dez 05 19:17:24 XperiaXZ3 usb_moded[3139]: Neither configured rndis0 nor fallback usb0 interface exists. Check your config!
Dez 05 19:17:24 XperiaXZ3 usb_moded[3139]: mode setting failed, try charging_only
Dez 05 19:17:24 XperiaXZ3 usb_moded[3139]: /config/usb_gadget/g1/functions/mass_storage.usb0: mkdir failed: No such file or directory
Dez 05 19:17:24 XperiaXZ3 usb_moded[3139]: function mass_storage.usb0 is not registered
Dez 05 19:17:24 XperiaXZ3 usb_moded[3139]: switch to charging mode failed
Dez 05 19:17:24 XperiaXZ3 usb_moded[3139]: failed to activate charging, all bets are off
Dez 05 19:17:24 XperiaXZ3 usb_moded[3139]: mode setting failed, fallback to undefined
Dez 05 19:17:24 XperiaXZ3 healthd: battery l=76 v=3932 t=24.5 h=2 st=3 c=388 fc=2971000 chg
```

Looks like 3.3 was broken already.

Maybe related: https://github.com/sailfishos-sony-tama/main/issues/70

**Comment by rinigus** [comment]
On 3.4.0.24, not working:

When set to Developer mode:

```
Dec 06 19:51:53 Sailfish mce[3135]: modules/battery-udev.c: mcebat_update(): charger_type: none -> usb
Dec 06 19:51:53 Sailfish usb_moded[3146]: /config/usb_gadget/g1/UDC: write failure: Value too large for defined data type
Dec 06 19:51:53 Sailfish kernel: ffs_data_put(): freeing
Dec 06 19:51:53 Sailfish usb_moded[3146]: Neither configured rndis0 nor fallback usb0 interface exists. Check your config!
Dec 06 19:51:53 Sailfish usb_moded[3146]: mode setting failed, try charging_only
Dec 06 19:51:53 Sailfish usb_moded[3146]: /config/usb_gadget/g1/functions/mass_storage.usb0: mkdir failed: No such file or directory
Dec 06 19:51:53 Sailfish usb_moded[3146]: function mass_storage.usb0 is not registered
Dec 06 19:51:53 Sailfish usb_moded[3146]: switch to charging mode failed
Dec 06 19:51:53 Sailfish usb_moded[3146]: failed to activate charging, all bets are off
Dec 06 19:51:53 Sailfish usb_moded[3146]: mode setting failed, fallback to undefined
```

When set to MTP
```
Dec 06 19:52:49 Sailfish mtp_service[26176]: [W] meegomtp1dot0::FSStoragePlugin::enumerateStorage_worker:496 - storage 65537 is ready
Dec 06 19:52:49 Sailfish mtp_service[26176]: [C] meegomtp1dot0::MTPTransporterUSB::activate:140 - MTPTransporterUSB::activate
Dec 06 19:52:49 Sailfish kernel: read descriptors
Dec 06 19:52:49 Sailfish kernel: read strings
Dec 06 19:52:49 Sailfish healthd: battery l=67 v=3862 t=25.2 h=2 st=3 c=780 fc=3217000 chg=
Dec 06 19:52:49 Sailfish usb_moded[3146]: /config/usb_gadget/g1/UDC: write failure: Value too large for defined data type
Dec 06 19:52:49 Sailfish systemd[3914]: Stopping MTP service...
Dec 06 19:52:49 Sailfish systemd[3914]: Stopped MTP service.
Dec 06 19:52:49 Sailfish usb_moded[3146]: mode setting failed, try charging_only
Dec 06 19:52:49 Sailfish usb_moded[3146]: /config/usb_gadget/g1/functions/mass_storage.usb0: mkdir failed: No such file or directory
Dec 06 19:52:49 Sailfish usb_moded[3146]: function mass_storage.usb0 is not registered
Dec 06 19:52:49 Sailfish usb_moded[3146]: switch to charging mode failed
Dec 06 19:52:49 Sailfish usb_moded[3146]: failed to activate charging, all bets are off
Dec 06 19:52:49 Sailfish usb_moded[3146]: mode setting failed, fallback to undefined
```

**Comment by rinigus** [comment]
In 3.2.1.20

When developer mode set, working
```
Dec 06 19:56:27 XperiaXZ2Compact usb_moded[13908]: unknown connection type reported, assuming disconnected
Dec 06 19:56:27 XperiaXZ2Compact mce[3205]: modules/battery-udev.c: mcebat_update(): charger_state: off -> on
Dec 06 19:56:27 XperiaXZ2Compact mce[3205]: tklock.c: tklock_datapipe_uiexception_type_cb(): uiexception_type = none -> notif
Dec 06 19:56:27 XperiaXZ2Compact kernel: water_detection wdet: wdet_check_connect_work: wdet_state=1, cable_timer=1
Dec 06 19:56:27 XperiaXZ2Compact usb_moded[13908]: unknown connection type reported, assuming disconnected
Dec 06 19:56:27 XperiaXZ2Compact mce[3205]: tklock.c: tklock_dbus_notification_beg_cb(): notification begin from name=:1.57 owner=:1.57 pid=7184 uid=100000 gid=999 priv=1 cmd=/usr/bin/lipstick -plugin evdevtouch:/dev/touchscreen -plugin e
Dec 06 19:56:27 XperiaXZ2Compact kernel: msm_pcm_routing_get_app_type_idx: App type not available, fallback to default
Dec 06 19:56:27 XperiaXZ2Compact kernel: usbpd usbpd0: Type-C Source (default) connected
Dec 06 19:56:27 XperiaXZ2Compact kernel: water_detection wdet: wdet_polling_set: state 0 -> 1
Dec 06 19:56:27 XperiaXZ2Compact kernel: water_detection wdet: wdet_polling_set: Polling stop
Dec 06 19:56:27 XperiaXZ2Compact kernel: msm-usb-ssphy-qmp 88e8000.ssphy: USB DP QMP PHY: Update TYPEC CTRL(2)
Dec 06 19:56:27 XperiaXZ2Compact kernel: msm-dwc3 a600000.ssusb: DWC3 exited from low power mode
Dec 06 19:56:27 XperiaXZ2Compact healthd: battery l=83 v=4001 t=27.2 h=2 st=3 c=243 fc=2520000 chg=
Dec 06 19:56:27 XperiaXZ2Compact healthd: battery l=83 v=4001 t=27.2 h=2 st=3 c=243 fc=2520000 chg=
Dec 06 19:56:27 XperiaXZ2Compact mce[3205]: modules/battery-udev.c: mcebat_update(): charger_type: none -> usb
Dec 06 19:56:27 XperiaXZ2Compact kernel: using random self ethernet address
Dec 06 19:56:27 XperiaXZ2Compact kernel: using random host ethernet address
Dec 06 19:56:27 XperiaXZ2Compact kernel: rndis0: HOST MAC 3e:cf:c1:5b:5f:fa
Dec 06 19:56:27 XperiaXZ2Compact kernel: rndis0: MAC 22:9d:fd:1f:9e:6c
Dec 06 19:56:27 XperiaXZ2Compact kernel: IPv6: ADDRCONF(NETDEV_UP): rndis0: link is not ready
Dec 06 19:56:27 XperiaXZ2Compact statefs[3228]: Technology (type= "gadget" ) tethering is  ""
Dec 06 19:56:27 XperiaXZ2Compact kernel: R0: [ifconfig][25264882149] [17:56:27.968067]  wlan: [19787:E :HDD] __wlan_hdd_ipv4_changed: 926: Adapter context is invalid 0000000000000000
Dec 06 19:56:27 XperiaXZ2Compact kernel: IPv6: ADDRCONF(NETDEV_UP): rndis0: link is not ready
Dec 06 19:56:28 XperiaXZ2Compact healthd: battery l=83 v=3988 t=27.2 h=2 st=3 c=390 fc=2520000 chg=
Dec 06 19:56:28 XperiaXZ2Compact kernel: msm-dwc3 a600000.ssusb: Avail curr from USB = 100
Dec 06 19:56:28 XperiaXZ2Compact healthd: battery l=83 v=3988 t=27.2 h=2 st=4 c=390 fc=2520000 chg=u
Dec 06 19:56:28 XperiaXZ2Compact kernel: android_work: sent uevent USB_STATE=CONNECTED
Dec 06 19:56:28 XperiaXZ2Compact kernel: android_work: sent uevent USB_STATE=DISCONNECTED
Dec 06 19:56:28 XperiaXZ2Compact estart[7701]: [W] unknown:163 - file:///usr/share/jolla-settings/pages/usb/usb.qml:163:13: QML Label: Binding loop detected for property "text"
Dec 06 19:56:28 XperiaXZ2Compact systemd[1]: Started udhcpcd DHCP server.
Dec 06 19:56:28 XperiaXZ2Compact audit[1]: SERVICE_START pid=1 uid=0 auid=4294967295 ses=4294967295 subj=kernel msg='unit=udhcpd comm="systemd" exe="/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
Dec 06 19:56:28 XperiaXZ2Compact unknown: type=1131 audit(1607277384.109:142): pid=1 uid=0 auid=4294967295 ses=4294967295 subj=kernel msg='unit=systemd-tmpfiles-clean comm="systemd" exe="/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
Dec 06 19:56:28 XperiaXZ2Compact unknown: type=1130 audit(1607277388.359:143): pid=1 uid=0 auid=4294967295 ses=4294967295 subj=kernel msg='unit=udhcpd comm="systemd" exe="/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
Dec 06 19:56:28 XperiaXZ2Compact udhcpd[19796]: udhcpd: started, v1.31.0
Dec 06 19:56:28 XperiaXZ2Compact mce[3205]: tklock.c: tklock_dbus_notification_beg_cb(): notification begin from name=:1.57 owner=:1.57 pid=7184 uid=100000 gid=999 priv=1 cmd=/usr/bin/lipstick -plugin evdevtouch:/dev/touchscreen -plugin e
Dec 06 19:56:28 XperiaXZ2Compact kernel: android_work: sent uevent USB_STATE=CONNECTED
Dec 06 19:56:28 XperiaXZ2Compact kernel: configfs-gadget gadget: high-speed config #1: b
Dec 06 19:56:28 XperiaXZ2Compact kernel: IPv6: ADDRCONF(NETDEV_CHANGE): rndis0: link becomes ready
Dec 06 19:56:28 XperiaXZ2Compact kernel: R0: [kworker/0:1][25273982810] [17:56:28.442060]  wlan: [87:E :HDD] __wlan_hdd_ipv6_changed: 232: Adapter context is invalid 0000000000000000
Dec 06 19:56:28 XperiaXZ2Compact kernel: msm-dwc3 a600000.ssusb: Avail curr from USB = 500
Dec 06 19:56:28 XperiaXZ2Compact kernel: android_work: sent uevent USB_STATE=CONFIGURED
```
When set to MTP, working
```
Dec 06 20:01:52 XperiaXZ2Compact usb_moded[13908]: unknown connection type reported, assuming disconnected
Dec 06 20:01:52 XperiaXZ2Compact mce[3205]: modules/battery-udev.c: mcebat_update(): charger_state: off -> on
Dec 06 20:01:52 XperiaXZ2Compact mce[3205]: tklock.c: tklock_datapipe_uiexception_type_cb(): uiexception_type = none -> notif
Dec 06 20:01:53 XperiaXZ2Compact usb_moded[13908]: unknown connection type reported, assuming disconnected
Dec 06 20:01:53 XperiaXZ2Compact mce[3205]: tklock.c: tklock_dbus_notification_beg_cb(): notification begin from name=:1.57 owner=:1.57 pid=7184 uid=100000 gid=999 priv=1 cmd=/usr/bin/lipstick -plugin evdevtouch:/dev/touchscreen -plugin e
Dec 06 20:01:53 XperiaXZ2Compact kernel: msm_pcm_routing_get_app_type_idx: App type not available, fallback to default
Dec 06 20:01:53 XperiaXZ2Compact kernel: usbpd usbpd0: Type-C Source (default) connected
Dec 06 20:01:53 XperiaXZ2Compact kernel: water_detection wdet: wdet_polling_set: state 0 -> 1
Dec 06 20:01:53 XperiaXZ2Compact kernel: water_detection wdet: wdet_polling_set: Polling stop
Dec 06 20:01:53 XperiaXZ2Compact kernel: msm-usb-ssphy-qmp 88e8000.ssphy: USB DP QMP PHY: Update TYPEC CTRL(2)
Dec 06 20:01:53 XperiaXZ2Compact kernel: msm-dwc3 a600000.ssusb: DWC3 exited from low power mode
Dec 06 20:01:53 XperiaXZ2Compact healthd: battery l=84 v=4030 t=27.5 h=2 st=3 c=281 fc=2520000 chg=
Dec 06 20:01:53 XperiaXZ2Compact kernel: CPU1: update max cpu_capacity 306
Dec 06 20:01:53 XperiaXZ2Compact healthd: battery l=84 v=4030 t=27.5 h=2 st=3 c=281 fc=2520000 chg=
Dec 06 20:01:53 XperiaXZ2Compact kernel: CPU2: update max cpu_capacity 306
Dec 06 20:01:53 XperiaXZ2Compact mce[3205]: modules/battery-udev.c: mcebat_update(): charger_type: none -> usb
Dec 06 20:01:53 XperiaXZ2Compact kernel: CPU0: update max cpu_capacity 306
Dec 06 20:01:53 XperiaXZ2Compact healthd: battery l=84 v=4030 t=27.5 h=2 st=3 c=281 fc=2520000 chg=
Dec 06 20:01:53 XperiaXZ2Compact su[21419]: pam_systemd(su-l:session): Using 600s D-Bus method call timeout
Dec 06 20:01:53 XperiaXZ2Compact systemd[1]: Started Session c6 of user nemo.
Dec 06 20:01:53 XperiaXZ2Compact su[21419]: pam_unix(su-l:session): session opened for user nemo by (uid=0)
Dec 06 20:01:53 XperiaXZ2Compact kernel: CPU2: update max cpu_capacity 306
Dec 06 20:01:53 XperiaXZ2Compact kernel: CPU4: update max cpu_capacity 1024
Dec 06 20:01:54 XperiaXZ2Compact healthd: battery l=84 v=4030 t=27.5 h=2 st=3 c=281 fc=2520000 chg=
Dec 06 20:01:54 XperiaXZ2Compact systemd[7088]: Started MTP service.
Dec 06 20:01:54 XperiaXZ2Compact su[21419]: pam_unix(su-l:session): session closed for user nemo
Dec 06 20:01:54 XperiaXZ2Compact healthd: battery l=84 v=4002 t=27.5 h=2 st=3 c=527 fc=2520000 chg=
Dec 06 20:01:54 XperiaXZ2Compact mtp_service[21462]: [W] meegomtp1dot0::MTPTransporterUSB::sendEvent:387 - event ignored - no active session
Dec 06 20:01:54 XperiaXZ2Compact mtp_service[21462]: [W] meegomtp1dot0::FSStoragePlugin::enumerateStorage_worker:491 - storage 65537 is ready
Dec 06 20:01:54 XperiaXZ2Compact mtp_service[21462]: [C] meegomtp1dot0::MTPTransporterUSB::activate:140 - MTPTransporterUSB::activate                                                                                                                     
Dec 06 20:01:54 XperiaXZ2Compact kernel: read descriptors
Dec 06 20:01:54 XperiaXZ2Compact kernel: read strings
Dec 06 20:01:54 XperiaXZ2Compact mtp_service[21462]: [W] ControlReaderThread::handleEvent:243 - Event:  BIND
Dec 06 20:01:54 XperiaXZ2Compact estart[7701]: [W] unknown:163 - file:///usr/share/jolla-settings/pages/usb/usb.qml:163:13: QML Label: Binding loop detected for property "text"
Dec 06 20:01:54 XperiaXZ2Compact mce[3205]: tklock.c: tklock_dbus_notification_beg_cb(): notification begin from name=:1.57 owner=:1.57 pid=7184 uid=100000 gid=999 priv=1 cmd=/usr/bin/lipstick -plugin evdevtouch:/dev/touchscreen -plugin e
Dec 06 20:01:54 XperiaXZ2Compact kernel: msm-dwc3 a600000.ssusb: Avail curr from USB = 100
Dec 06 20:01:54 XperiaXZ2Compact kernel: water_detection wdet: wdet_polling_set: state 1 -> 1
Dec 06 20:01:54 XperiaXZ2Compact kernel: android_work: sent uevent USB_STATE=CONNECTED
Dec 06 20:01:54 XperiaXZ2Compact kernel: android_work: sent uevent USB_STATE=DISCONNECTED
Dec 06 20:01:54 XperiaXZ2Compact healthd: battery l=84 v=4002 t=27.5 h=2 st=4 c=527 fc=2520000 chg=u
Dec 06 20:01:54 XperiaXZ2Compact kernel: android_work: sent uevent USB_STATE=CONNECTED
Dec 06 20:01:54 XperiaXZ2Compact mtp_service[21462]: [W] ControlReaderThread::handleEvent:243 - Event:  ENABLE
Dec 06 20:01:54 XperiaXZ2Compact kernel: configfs-gadget gadget: high-speed config #1: b
Dec 06 20:01:54 XperiaXZ2Compact kernel: msm-dwc3 a600000.ssusb: Avail curr from USB = 500
Dec 06 20:01:54 XperiaXZ2Compact kernel: android_work: sent uevent USB_STATE=CONFIGURED
Dec 06 20:01:54 XperiaXZ2Compact healthd: battery l=84 v=4002 t=27.5 h=2 st=4 c=527 fc=2520000 chg=u
Dec 06 20:01:55 XperiaXZ2Compact healthd: battery l=84 v=4002 t=27.5 h=2 st=4 c=527 fc=2520000 chg=u
Dec 06 20:01:55 XperiaXZ2Compact healthd: battery l=84 v=4002 t=27.5 h=2 st=4 c=527 fc=2520000 chg=u
Dec 06 20:01:55 XperiaXZ2Compact healthd: battery l=84 v=4002 t=27.5 h=2 st=2 c=527 fc=2520000 chg=u
Dec 06 20:01:55 XperiaXZ2Compact healthd: battery l=84 v=4002 t=27.5 h=2 st=2 c=527 fc=2520000 chg=u
Dec 06 20:01:55 XperiaXZ2Compact mce[3205]: modules/battery-udev.c: mcebat_update(): battery_state: discharging -> charging
```
```
```

**Comment by rinigus** [comment]
On mal's advice, fixed by updating usb-moded to mer/0.86.0+mer51
